datumaro/util/image.py

```diff
@@ -199,28 +197,8 @@ def decode_image(image_bytes: bytes,
 
 def find_images(dirpath: str, exts: Union[str, Iterable[str]] = None,
         recursive: bool = False, max_depth: int = None) -> Iterator[str]:
-    if isinstance(exts, str):
-        exts = {'.' + exts.lower().lstrip('.')}
-    elif exts is None:
-        exts = IMAGE_EXTENSIONS
-    else:
-        exts = {'.' + e.lower().lstrip('.') for e in exts}
-
-    def _check_image_ext(filename: str):
-        dotpos = filename.rfind('.')
-        if 0 < dotpos: # exclude '.ext' cases too
-            ext = filename[dotpos:].lower()
-            if ext in exts:
-                return True
-        return False
-
-    for d, _, filenames in walk(dirpath,
-            max_depth=max_depth if recursive else 0):
-        for filename in filenames:
-            if not _check_image_ext(filename):
-                continue
-
-            yield osp.join(d, filename)
+    yield from find_files(dirpath, exts=exts or IMAGE_EXTENSIONS,
```

Comment:
```suggestion
    return find_files(dirpath, exts=exts or IMAGE_EXTENSIONS,
```

This will avoid creating a new generator for the `find_images` function call.